src/payload.ts

```diff
@@ -160,7 +160,7 @@ export class BasePayload<TGeneratedTypes extends GeneratedTypes> {
       );
     }
 
-    if (options.mongoURL !== false && typeof options.mongoURL !== 'string') {
+    if (!options.local && options.mongoURL !== false && typeof options.mongoURL !== 'string') {
```

Comment:
@denolfe This line broke the build because it is trying to connect to the DB for the schema generation. What was the intention here?

Reply:
I believe this was introduced to handle passing a db adapter in the config. We can look at removing it, but I know it needed at one point.